packages/cozy-stack-client/src/CozyStackClient.js

```diff
@@ -63,22 +64,22 @@ class CozyStackClient {
   async fetch(method, path, body, opts = {}) {
     const options = { ...opts }
     options.method = method
-    const headers = (options.headers = options.headers || {})
+    const headers = (options.headers = { ...opts.headers })
 
     if (method !== 'GET' && method !== 'HEAD' && body !== undefined) {
       if (headers['Content-Type']) {
         options.body = body
       }
     }
 
-    const credentials = options.credentials || this.getCredentials()
-    if (credentials) {
-      headers['Authorization'] = credentials
-      // the option credentials:include tells fetch to include the cookies in the
-      // request even for cross-origin requests
-      options.credentials = 'include'
+    if (!headers.Authorization) {
+      headers.Authorization = this.getAuthorizationHeader()
     }
 
+    // the option credentials:include tells fetch to include the cookies in the
+    // request even for cross-origin requests
+    options.credentials = 'include'
```

Comment:
Do we need this ? as we are using the Authorization header, the cookie is probably not necessary anyways.

Reply:
Some API from the stack necessitate to have a session cookie. Until recently it was for example necessary for the registry icons. I agree that this may not be suitable but we would need to do an audit of the stack routes to know this. Maybe it is needed for the auth routes.

I think we should tackle this in another issue.